Userland/Libraries/LibJS/Runtime/Array.cpp

```diff
@@ -81,14 +81,6 @@ JS_DEFINE_NATIVE_GETTER(Array::length_getter)
     if (!this_object)
         return {};
 
-    // TODO: could be incorrect if receiver/this_value is fixed or changed
-    if (!this_object->is_array()) {
-        auto value = this_object->internal_get(vm.names.length.to_string_or_symbol(), this_object);
-        if (vm.exception())
-            return {};
-        return value;
-    }
-
```

Comment:
This seems unrelated, perhaps should be its own commit with an explanation?

Reply:
It is actually related, but ill split it into its own commit anyways (This is a matching removal of the same block in the setter, which used `define_property` but instead of replacing the define_property i removed both blocks as they are no longer relevant after the object rewrite.

Reply:
fixed